docs/INTRO.md

```diff
@@ -0,0 +1,441 @@
+
+# Untyped Plutus Core (UPLC)
+
+Plutarch compiles to UPLC. Most Plutarch end-users will not need to concern themselves with the details of UPLC, but a brief overview will aid in building a mental model of how Plutarch works.
+
+Unlike Haskell, UPLC is a low-level and untyped language implementing a basic lambda calculus. Consequently, it supports only a handful of built-in values and functions which may be strung together in lambda applications. The built-in types provided by UPLC include the usual primitive types -- integers, byte strings and strings, booleans, and so forth -- and a special `Data` value that can encode representations of arbitrary sum-of-products Haskell types.
+
+While the _semantic_ meaning of a Haskell type such as `Maybe Integer` is missing in UPLC, it still can be _represented_ as `Data` in UPLC. UPLC doesn't track what differentiates semantically distinct `Data` values and will not prevent a programmer from operating on the `Data` representation in non-sensical ways. 
+
+Plutarch's solution is to _tag_ scripts that compile to UPLC (i.e., Plutarch ` Term`s) with types. Doing so allows the Plutarch compiler to track and type check operations on semantically distinct `Data` values. These tags are provided by "Plutarch Types", or "types of kind `PType`". 
+
+For the Plutarch compiler to bridge between arbitrary, semantically-rich Haskell types and the untyped `Data` of UPLC, it is necessary to associate various bits of information with `PType`s. On the one hand, each `PType` should have some semantic, type-level richness such as typeclass instances (otherwise, there would be little point in programming in Haskell!). On the other hand, each `PType` needs to have a UPLC representation, either as a built-in primitive value or as `Data`, to compile to UPLC.
+
+# Plutarch Types
+
+When this guide uses the term "Plutarch Type," we explicitly talk about a type of _kind_ `PType`. We will refer to  _" types of kind `PType` "_ simply as" `PType`s." We explicitly qualify when referring to the _kind_ `PType`.
+
+`PType` is defined as `type PType = S -> Type`; that is, it is a _kind synonym_ for `S -> Type` (where `S` and `Type` are themselves kinds). This synonym is important to keep in mind because when querying the kind of something like `PBool` in, say, GHCi, we will _not_ see `PType` as the kind. Instead, we get 
+
+```hs
+ghci> :k PBool
+PBool :: S -> Type
+```
+
+Thus, any time we see the kind `S -> Type`, we should mentally substitute its kind synonym `PType`. We reiterate: types of kind `PType`, should be considered as _tags_ on computation. They do not represent types of values in the same way as standard Haskell types. 
+
+The kind of basic types such as `Integer` in Haskell is `*`; the corresponding "basic" kind in Plutarch is simply `PType`. Higher-kinded types in Haskell, such as `Maybe`, will kinds such as `* -> *`. In Plutarch, the corresponding kind is:
```

Comment:
This is a good call.  Is there actually a difference between `Type` and `*`, or are they synonyms? I hadn't encountered the `Type` kind before MLabs, so I think it may be worthwhile to mention whether or not there's a distinction.

Reply:
They are one and the same. `*` is Haskell 2010's way of pronouncing `Type`. Times have changed a lot since then: https://ghc.gitlab.haskell.org/ghc/doc/users_guide/exts/poly_kinds.html#the-kind-type

Reply:
Got it. I've added a blockqutoe note with this info and a link to basic type level programming concepts. I found it a helpful post to skim when I was trying to get up to speed on these things a few months ago.